src/gfx/egui/src/input.rs

```
@@ -0,0 +1,267 @@
+        CursorIcon::Progress => egui_CursorIcon_Progress,
+        CursorIcon::Wait => egui_CursorIcon_Wait,
+        CursorIcon::Cell => egui_CursorIcon_Cell,
+        CursorIcon::Crosshair => egui_CursorIcon_Crosshair,
+        CursorIcon::Text => egui_CursorIcon_Text,
+        CursorIcon::VerticalText => egui_CursorIcon_VerticalText,
+        CursorIcon::Alias => egui_CursorIcon_Alias,
+        CursorIcon::Copy => egui_CursorIcon_Copy,
+        CursorIcon::Move => egui_CursorIcon_Move,
+        CursorIcon::NoDrop => egui_CursorIcon_NoDrop,
+        CursorIcon::NotAllowed => egui_CursorIcon_NotAllowed,
+        CursorIcon::Grab => egui_CursorIcon_Grab,
+        CursorIcon::Grabbing => egui_CursorIcon_Grabbing,
+        CursorIcon::AllScroll => egui_CursorIcon_AllScroll,
+        CursorIcon::ResizeHorizontal => egui_CursorIcon_ResizeHorizontal,
+        CursorIcon::ResizeNeSw => egui_CursorIcon_ResizeNeSw,
+        CursorIcon::ResizeNwSe => egui_CursorIcon_ResizeNwSe,
+        CursorIcon::ResizeVertical => egui_CursorIcon_ResizeVertical,
+        CursorIcon::ResizeEast => egui_CursorIcon_ResizeEast,
+        CursorIcon::ResizeSouthEast => egui_CursorIcon_ResizeSouthEast,
+        CursorIcon::ResizeSouth => egui_CursorIcon_ResizeSouth,
+        CursorIcon::ResizeSouthWest => egui_CursorIcon_ResizeSouthWest,
+        CursorIcon::ResizeWest => egui_CursorIcon_ResizeWest,
+        CursorIcon::ResizeNorthWest => egui_CursorIcon_ResizeNorthWest,
+        CursorIcon::ResizeNorth => egui_CursorIcon_ResizeNorth,
+        CursorIcon::ResizeNorthEast => egui_CursorIcon_ResizeNorthEast,
+        CursorIcon::ResizeColumn => egui_CursorIcon_ResizeColumn,
+        CursorIcon::ResizeRow => egui_CursorIcon_ResizeRow,
+        CursorIcon::ZoomIn => egui_CursorIcon_ZoomIn,
+        CursorIcon::ZoomOut => egui_CursorIcon_ZoomOut,
+    }
+}
+
+pub fn translate_raw_input(
+    input: &egui_Input,
+) -> Result<egui::RawInput, Box<dyn std::error::Error>> {
```

```suggestion
) -> Result<egui::RawInput, &str> {
```

---

I changed it to 
```
pub enum TranslationError {
    Text(std::str::Utf8Error),
}
```
you should be able to silently ignore that one in the UI shard though, maybe just a log